treeherder/perf/management/commands/import_perf_data.py

```diff
@@ -1,29 +1,72 @@
+import datetime
 from optparse import make_option
 from urlparse import urlparse
 
 import concurrent.futures
 from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
 
 from treeherder.client import PerfherderClient, PerformanceTimeInterval
-from treeherder.model.derived.jobs import JobsModel
-
-
-def _add_series(server_params, project, time_intervals, signature_hash,
-                signature_props, mysql_debug, verbose):
-    with JobsModel(project) as jm:
-        jm.DEBUG = mysql_debug
-        if verbose:
-            print(signature_hash)
-
-        jm.set_series_signature(signature_hash, signature_props)
-        for time_interval in time_intervals:
-            pc = PerfherderClient(protocol=server_params.scheme,
-                                  host=server_params.netloc)
-            series = pc.get_performance_series(project, signature_hash,
-                                               time_interval=time_interval)
-            jm.store_performance_series(time_interval, 'talos_data',
-                                        str(signature_hash),
-                                        series)
+from treeherder.model.models import (MachinePlatform,
+                                     OptionCollection,
+                                     PerformanceFramework,
+                                     PerformanceSignature,
+                                     PerformanceDatum, Repository)
+
+
+def _add_series(server_params, project_name, signature_hash, signature_props, verbose):
+    if verbose:
+        print signature_hash
+    if not PerformanceSignature.objects.filter(uuid=signature_hash):
```

Comment:
Please use PerformanceSignature.objects.get and except PerformanceSignature.DoesNotExist to catch the case where you have to create it